docs/website/contents/for-developers/ProtocolVersioning.md

```diff
@@ -0,0 +1,155 @@
+- Transparent Rung (top).
+
+### The Cardano Message Data Onion
+
+The Message Data Rung is organized as the layers of an onion.
+So are some other rungs, but their structure is beyond the scope of this document.
+
+- [Ledger Layer](https://github.com/IntersectMBO/cardano-ledger/).
+    - The notion of _era_ is an organizing concept within the Ledger Layer --- more on this below.
+    - For each era, the content of its blocks, transactions, queries, and query responses.
+- [Consensus Layer](https://github.com/IntersectMBO/ouroboros-consensus/).
+    - The content of the few queries and their responses that either do not depend on the ledger era or else regard when prior era transitions happened, which the Ledger Layer does not record.
+    - An additional envelope wrapping block headers, blocks, transactions, era-specific queries, and era-specific query responses.
+      The envelope indicates a particular era via an _era tag_ [^era-tag-parse], which affects the validity of the object but not its identifying hash.
+- [Network Layer](https://github.com/IntersectMBO/ouroboros-network) (outermost) [^mux-layer].
+    - ChainSync, BlockFetch, TxSubmission, LocalStateQuery, etc; see "Chapter 3 Mini Protocols" within the [Ouroboros Network Specification](https://ouroboros-network.cardano.intersectmbo.org/pdfs/network-spec/network-spec.pdf) [^network-pdf-link].
+    - The content of each mini protocol message, excluding the payload.
+
+### Precise Answer
+
+The protocol version must be incremented if an old node and a new node would otherwise disagree on any of the following.
+
+- *Chain Validity Predicates*, the validity predicates within the Message Validity Rung for block headers and blocks.
+- *Chain Order*, the [partial order](https://en.wikipedia.org/wiki/Partially_ordered_set) within the Message Validity Rung that determines one chain is better than another.
+  Tiebreakers are a gray area, since Praos does not inherently rely on them.
+- *Chain Data Model*, the parts of the Message Data Rung used to define the Chain Validity Predicates and the Chain Order.
+    - A change to the data model of the node's maintained state --- as opposed to its messages --- would not inherently trigger this.
+    - However, if such a change were meaningful (ie not just a refactor/optimization/etc), then some Chain Validity Predicates would necessarily also change.
+- *Chain Codecs*, the parts of the mapping from the CBOR Rung to the Chain Data Model that affect the hashes therein.
+  Two nodes won't even be able to exchange transactions, block headers, and/or blocks if they disagree on their identifying hashes.
+
+The most common change in Cardano's protocol so far has been the addition of a new kind/variant of transaction, which triggers all of these except the Chain Order.
+The Chain Order has only changed from Praos's length to Ouroboros Genesis's length-soon-after-the-intersection, which is slight.
```

Comment:
```suggestion
The Chain Order has only changed from Praos's length to Ouroboros Genesis's density-after-the-intersection, which is slight.
```

I would not relate Genesis and length artificially. Genesis is about density.

Reply:
I'm hung up on this a little, since `length-soon-after-the-intersection` _is_ density, assuming "soon" is a fixed value.